### PR TITLE
Implement tyeps on client-side

### PR DESCRIPTION
**Pull request by nabijaczleweli**
Should we stay with `browserify` or use a module system?
If `browserify`, should we force-include `victor` and `keypress`?

### CODE REVIEW

File: client/src/game.ts [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+    constructor() {
+        document.addEventListener('DOMContentLoaded', this.onDOMReady);
+    }
+
+    //
+    // DOM events
+    //
+
+    onDOMReady = (): void => {
```
Comment (EthanSteinberg) [inline client/src/game.ts:23]: Minor annoyance, these should probably be methods.

File: client/src/gamescreen.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+export class GameScreen {
+    public state: MessageData.WorldState;
+    playerId: number;
+
+    // Current mouse position relative to canvas origin.
+    // For aim vector rendering.
+    mousePos: MousePos = null;
+
+    keymap: Array<boolean> = [];
```
Comment (EthanSteinberg) [inline client/src/gamescreen.ts]: Key map should be an Object or Map, not an Array but it doesn't really matter.

File: client/src/game.ts (hunk 1 above)
Comment (nabijaczleweli) [inline client/src/game.ts:23]: nah, scoping is a bitch, see https://github.com/LoungeCPP/Tatsoryk/pull/44/commits/263a259232f72604256403c052beba2f0c2c1042

**Comment by EthanSteinberg** [comment]
It looks good. I only had two minor comments. :+1: Nice work.

File: client/src/gamescreen.ts (hunk 2 above)
Comment (nabijaczleweli) [inline client/src/gamescreen.ts]: Done

**Pull request merged**